packages/apollo-federation/src/service/__tests__/buildFederatedSchema.test.ts

```diff
@@ -10,40 +17,88 @@ const EMPTY_DOCUMENT = {
   definitions: [],
 };
 
-describe('buildFederatedSchema', () => {
-  it(`should mark a type with a key field as an entity`, () => {
-    const schema = buildFederatedSchema(gql`
-      type Product @key(fields: "upc") {
-        upc: String!
-        name: String
-        price: Int
-      }
-    `);
+const testFederatedSchema = (
+  name: string,
+  createSchema: () => GraphQLSchema,
+  runTest: (schema: GraphQLSchema) => Promise<void>,
+) => {
+  const schema = createSchema();
+
+  it(name, async () => {
+    await runTest(schema);
+  });
+
+  it(`${name} (using "schema" argument)`, async () => {
+    await runTest(buildFederatedSchema(schema));
```

Comment:
This expands to `runTest(buildFedaratedSchema(buildFedaratedSchema(gql(...))))`. Why the double call? Is this ensuring `buildFedaratedSchema` is idempotent?

Reply:
It was a quick way for me to test passing a schema into the build call and have all the same tests pass. One call is created using gql and one uses the schema object created from the same call, insuring they create the same schema. I can recreate this using a real schema object though. Just more work for a PR i wasn’t sure would make it in.

Reply:
Okay, so I just got home from being out of town to look into this more.  Yes, the double call is ensuring Idempotency between the schema created, and the second call, since it's an instance of `GraphQLSchema`, runs through the new code.  It's simply a way to re-use all `buildFederatedSchema` tests with both `buildFederatedSchema(sdl: (GraphQLSchemaModule | DocumentNode)[] | DocumentNode)` and `buildFederatedSchema(schema: GraphQLSchema)`.